### PR TITLE
[FIX] account: avoid infinite loop when creating taxes

### PR DESCRIPTION
**Pull request by jjmaksoud**
When updating taxes, creation of taxes of type 'group' does not
work correctly if either a child or the parent itself has been deleted.
Indeed the creation of _generate_tax() method suppose that both the parent
and the child needs to be created in the same batch while we can have
databases where one of those already exists and will go in the 'update'
and not 'creation' part of the script.

opw-3444084
opw-3440590
opw-3435425
opw-3415544
[edit avg-odoo]: opw-3452532

Description of the issue/feature this PR addresses:

**Current behavior before PR:**
During an upgrade, if a database has a tax template with computation type 'group', and the template's tax is edited or removed, but at least one of the children taxes not edited, then the upgrade will get stuck in an infinite iteration waiting for the child taxes to be created.

**Desired behavior after PR is merged:**
The child taxes that dont need to be created are provided to the loop so the parent can be created.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by clbr-odoo** [comment]
@jjmaksoud Thank you for your work ! 
I allowed myself to make some changes to simplify a bit and added tests.
I'm sending it for r+ :slightly_smiling_face:

File: addons/account/models/chart_template.py [hunk 1]
```diff
@@ -1219,6 +1225,10 @@ def _generate_tax(self, company, accounts_exist=False):
                     'cash_basis_transition_account_id': template.cash_basis_transition_account_id.id,
                 }
 
+                for existing_template, existing_tax in existing_template_to_tax:
+                    if template in existing_template.children_tax_ids and tax not in existing_tax:
```
Comment (oco-odoo) [inline addons/account/models/chart_template.py:1229]: Shouldn't it rather be something like this ?

```suggestion
                    if template in existing_template.children_tax_ids and tax not in existing_tax.children_tax_ids:
```

File: addons/account/models/chart_template.py [hunk 2]
```diff
@@ -1177,10 +1179,12 @@ def _get_tax_vals_complete(self, company, tax_template_to_tax):
         })
         return vals
 
-    def _generate_tax(self, company, accounts_exist=False):
+    def _generate_tax(self, company, accounts_exist=False, existing_template_to_tax=None):
         """ This method generate taxes from templates.
 
             :param company: the company for which the taxes should be created from templates in self
+            :account_exist: are accounts already created
```
Comment (oco-odoo) [inline addons/account/models/chart_template.py:1186]:
```suggestion
            :account_exist: whether accounts have already been created
```

**Review by oco-odoo — CHANGES_REQUESTED** [review]

File: addons/account/models/chart_template.py (hunk 1 above)
Comment (clbr-odoo) [reply addons/account/models/chart_template.py:1229]: Of course indeed. Thanks !

**Review by oco-odoo — APPROVED** [review]
@robodoo r+
Thanks :)

**Pull request closed without merging**